cpp/src/arrow/compute/kernels/vector_replace_test.cc

```diff
@@ -798,5 +828,688 @@ TYPED_TEST(TestReplaceBinary, ReplaceWithMaskRandom) {
+                            this->array(R"(["afd", "afd", "afd", "afd"])"));
+  this->AssertFillNullArray(FillBackwardNull, this->array(R"([null, "afd", null])"),
+                            this->array(R"(["afd", "afd", null])"));
+  this->AssertFillNullArray(FillBackwardNull, this->array(R"([null, null, "afd", null])"),
+                            this->array(R"(["afd", "afd", "afd", null])"));
+  this->AssertFillNullArray(FillBackwardNull,
+                            this->array(R"([null, null, null, "afd", null])"),
+                            this->array(R"(["afd", "afd", "afd", "afd", null])"));
+  this->AssertFillNullArray(FillBackwardNull,
+                            this->array(R"([null, null, "afd",null, "qwe", null])"),
+                            this->array(R"(["afd", "afd", "afd", "qwe", "qwe", null])"));
+
+  this->AssertFillNullArray(FillBackwardNull, this->array(R"(["tyu", "afd", null])"),
+                            this->array(R"(["tyu", "afd", null])"));
+  this->AssertFillNullArray(FillBackwardNull,
+                            this->array(R"(["tyu", "afd", null, null, null, null])"),
+                            this->array(R"(["tyu", "afd" ,null, null, null, null])"));
+  this->AssertFillNullArray(FillBackwardNull,
+                            this->array(R"(["tyu", "afd", null, "qwe", null, null])"),
+                            this->array(R"(["tyu", "afd" , "qwe", "qwe", null, null])"));
+  this->AssertFillNullArray(
+      FillBackwardNull,
+      this->array(R"(["tyu", "afd", null, "qwe", null, null, "oiutyu"])"),
+      this->array(R"(["tyu", "afd" ,"qwe", "qwe", "oiutyu", "oiutyu", "oiutyu"])"));
+  this->AssertFillNullArray(
+      FillBackwardNull, this->array(R"(["tyu", "afd", null, "qwe", null, null, "qwe"])"),
+      this->array(R"(["tyu", "afd" ,"qwe" , "qwe", "qwe", "qwe", "qwe"])"));
+  this->AssertFillNullArray(
+      FillBackwardNull,
+      this->array(R"(["tyu", "afd", null, "qwe", null, "oiutyu", null])"),
+      this->array(R"(["tyu", "afd" ,"qwe" , "qwe", "oiutyu", "oiutyu", null])"));
+  this->AssertFillNullArray(
+      FillBackwardNull,
+      this->array(R"(["tyu", "afd", null, "qwe", null, "oiutyu", "aaaagggbbb"])"),
+      this->array(R"(["tyu", "afd" ,"qwe", "qwe", "oiutyu", "oiutyu", "aaaagggbbb"])"));
+  this->AssertFillNullArray(
+      FillBackwardNull,
+      this->array(R"(["tyu", "afd" ,"qwe", "qwe", "oiutyu", "oiutyu", "aaaagggbbb"])"),
+      this->array(R"(["tyu", "afd" ,"qwe", "qwe", "oiutyu", "oiutyu", "aaaagggbbb"])"));
+}
+
+// For Test Blocks
+TYPED_TEST(TestFillNullNumeric, FillNullForwardLargeInput) {
+  using CType = typename TypeTraits<TypeParam>::CType;
+  random::RandomArrayGenerator rand(/*seed=*/1000);
+  int64_t len_null = 500;
+  int64_t len_random = 1000;
+  std::shared_ptr<Array> array_random =
+      rand.Numeric<TypeParam>(len_random, /*min=*/0, /*max=*/200, /*nulls=*/0);
+
+  if (array_random) {
```

Comment:
Why the check here?

Reply:
Ditto below.

Reply:
Ping here.

Reply:
Just a reminder here.

Reply:
@AlvinJ15: what is the reason for this check? If there is one, please let me know - I'm just wondering why it's here.

Reply:
same as the next comment, no support for rand.Numeric( ) for DateType, even if I use rand.ArrayOf no support for ConstantArrayGenerator::Numeric(DateTypes).